adapters/relevantdigital/relevantdigital.go

```diff
@@ -0,0 +1,278 @@
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+	name     string
+}
+
+const RELEVANT_DOMAIN = ".relevant-digital.com"
+const DEFAULT_TIMEOUT = 1000
+const DEFAULT_BUFFER_MS = 250
+const STORED_REQUEST_EXT = "{\"prebid\":{\"debug\":%t,\"storedrequest\":{\"id\":\"%s\"}},\"relevant\":{\"count\":%d,\"adapterType\":\"server\"}}"
+const STORED_IMP_EXT = "{\"prebid\":{\"storedrequest\":{\"id\":\"%s\"}}}"
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+	return &adapter{
+		endpoint: template,
+		name:     bidderName.String(),
+	}, nil
+}
+
+func patchBidRequestExt(prebidBidRequest *openrtb2.BidRequest, id string) error {
+	count, cerr := jsonparser.GetInt(prebidBidRequest.Ext, "relevant", "count")
+	if cerr != nil {
+		count = 0
+	}
+
+	if count >= 5 {
+		return &errortypes.FailedToRequestBids{
+			Message: "too many requests",
+		}
+	} else {
+		count = count + 1
+	}
+
+	debug, derr := jsonparser.GetBoolean(prebidBidRequest.Ext, "prebid", "debug")
+	if derr != nil {
+		debug = false
+	}
+
+	prebidBidRequest.Ext = []byte(fmt.Sprintf(STORED_REQUEST_EXT, debug, id, count))
+	return nil
+}
+
+func patchBidImpExt(imp *openrtb2.Imp, id string) {
+	imp.Ext = []byte(fmt.Sprintf(STORED_IMP_EXT, id))
+	if imp.Banner != nil {
+		imp.Banner.Ext = nil
+	}
+	if imp.Video != nil {
+		imp.Video.Ext = nil
+	}
+	if imp.Native != nil {
+		imp.Native.Ext = nil
+	}
+	if imp.Audio != nil {
+		imp.Audio.Ext = nil
+	}
+}
+
+func setTMax(prebidBidRequest *openrtb2.BidRequest, pbsBufferMs int) {
+	timeout := float64(prebidBidRequest.TMax)
+	if timeout <= 0 {
+		timeout = DEFAULT_TIMEOUT
+	}
+	buffer := float64(pbsBufferMs)
+	prebidBidRequest.TMax = int64(math.Min(math.Max(timeout-buffer, buffer), timeout))
+}
+
+func cloneBidRequest(prebidBidRequest *openrtb2.BidRequest) (*openrtb2.BidRequest, error) {
+	jsonRes, err := json.Marshal(prebidBidRequest)
+	if err != nil {
+		return nil, err
+	}
+	var copy openrtb2.BidRequest
+	err = json.Unmarshal(jsonRes, &copy)
+	return &copy, err
+}
+
+func createBidRequest(prebidBidRequest *openrtb2.BidRequest, params []*openrtb_ext.ExtRelevantDigital) (*openrtb2.BidRequest, error) {
+	bidRequestCopy, err := cloneBidRequest(prebidBidRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	err = patchBidRequestExt(bidRequestCopy, params[0].AccountId)
+	if err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("failed to create bidRequest, error: %s", err),
+		}
+	}
+
+	setTMax(bidRequestCopy, params[0].PbsBufferMs)
```

Comment:
only first index value is considered for `AccountId` and `PbsBufferMs`. 

What if request has different `AccountId` and `PbsBufferMs` values for imp in the incomming request? If only first index value is considered for `AccountId` and `PbsBufferMs` then should explicity mention this behaviour in bidder docs.

Reply:
This is correct, setting different config values on different imps is not supported, as described in the documentation: 

`Setting different global config-parameters on different bids is not supported in general` [0]

[0] https://docs.prebid.org/dev-docs/bidders/relevantdigital.html